src/main/java/org/apache/commons/lang3/builder/ReflectionDiffBuilder.java

```diff
@@ -110,8 +110,9 @@ private void appendFields(final Class<?> clazz) {
         for (final Field field : FieldUtils.getAllFields(clazz)) {
             if (accept(field)) {
                 try {
-                    diffBuilder.append(field.getName(), readField(field, left, true),
-                            readField(field, right, true));
+                    Object leftObject = readField(field, left, true);
```

Comment:
Is this change needed for the type changes? Please revert it otherwise.

Reply:
Please read the description:

> There's a case as shown in `ReflectionDiffBuilder.java` that if the type inference results in ambiguity, it may cause compile error.

Reply:
@tisonkun 
I still don't get it, why is this needed in this PR as it relates to the generics change?

Reply:
<img width="1056" alt="image" src="https://user-images.githubusercontent.com/18818196/192087964-342f0e2d-57ab-4d45-8ade-51aa183c1482.png">

Reply:
```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/chenzili/Brittani/commons-lang/src/main/java/org/apache/commons/lang3/builder/ReflectionDiffBuilder.java:[113,32] reference to append is ambiguous
  both method append(java.lang.String,short[],short[]) in org.apache.commons.lang3.builder.DiffBuilder and method append(java.lang.String,java.lang.Object[],java.lang.Object[]) in org.apache.commons.lang3.builder.DiffBuilder match
```

Reply:
Revert this change, please.

Reply:
Revert this change will cause compile to fail. I can do it and you see what happens.

Reply:
Reverted at bc4e02229407a80b1315c8c5f00cad859724aa4e.

Now the compilation failed: https://github.com/apache/commons-lang/actions/runs/3210472599/jobs/5247938888.

Reply:
Does that mean that existing call sites in users' applications will also fail?

Reply:
If you think it breaks the existing code and should not do it, you can reject the PR. But I'm a bit confused since reverting it obviously causes compile to fail.